packages/compat/src/v1-app.ts

```diff
@@ -184,55 +183,35 @@ export default class V1App implements V1Package {
 
   @Memoize()
   babelConfig(): TransformOptions {
-    let plugins = get(this.app.options, 'babel.plugins') as any[];
-    if (!plugins) {
-      plugins = [];
-    } else {
-      plugins = plugins.filter(p => {
-        // even if the app was using @embroider/macros, we drop it from the config
-        // here in favor of our globally-configured one.
-        return (
-          !isEmbroiderMacrosPlugin(p) &&
-          // similarly, if the app was already using
-          // ember-cli-htmlbars-inline-precompile, we remove it here because we
-          // have our own always-installed version of that (v2 addons are
-          // allowed to assume it will be present in the final app build, the
-          // app doesn't get to turn that off or configure it.).
-          !TemplateCompiler.isInlinePrecompilePlugin(p)
-        );
-      });
-    }
+    const babelAddon = (this.app.project as any).findAddonByName('ember-cli-babel');
+    const babelConfig = babelAddon.buildBabelOptions({
+      'ember-cli-babel': {
+        includeExternalHelpers: true,
+        compileModules: false,
+        disableDebugTooling: true,
+        disablePresetEnv: false,
+        disableEmberModulesAPIPolyfill: false,
+        disableDecoratorTransforms: false,
+      },
+    });
 
-    // this is reproducing what ember-cli-babel does. It would be nicer to just
-    // call it, but it require()s all the plugins up front, so not serializable.
-    // In its case, it's mostly doing it to set basedir so that broccoli caching
-    // will be happy, but that's irrelevant to us here.
+    let plugins = babelConfig.plugins as any[];
+    let presets = babelConfig.presets;
+
+    // install embroider macros
```

Comment:
This comment is wrong. `debugMacrosPlugin` is a stock ember-cli, it's not related to embroider macros.

Reply:
oops